crates/typst-library/src/diag.rs

```diff
@@ -568,31 +571,270 @@ impl From<PackageError> for EcoString {
     }
 }
 
+pub type LoadResult<T> = Result<T, LoadError>;
+
+/// A call site independent error that occurred during data loading.
+/// This avoids polluting the memoization with [`Span`]s and [`FileId`]s from source files.
+/// Can be turned into a [`SourceDiagnostic`] using the [`LoadedAt::in_text`]
+/// or [`LoadedAt::in_invalid_text`] methods available on [`LoadResult`].
+///
+/// [`FileId`]: typst_syntax::FileId
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct LoadError {
```

Comment:
`LoadError` is currently 56 bytes large (on 64 bit architectures), maybe it should be internally wrapped in an `Arc`?